verifier.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ratify
+
+import (
+	"context"
+	"fmt"
+
+	oci "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/ratify-project/ratify-go/errors"
+	"github.com/ratify-project/ratify-go/internal/common"
+)
+
+// RegisteredVerifiers saves the registered verifier factories.
+var RegisteredVerifiers = make(map[string]func(config VerifierConfig) (Verifier, error))
+
+// Verifier is an interface that defines methods to verify an artifact associated
+// with a subject.
+type Verifier interface {
+	// Name returns the name of the verifier
+	Name() string
+
+	// Type returns the type name of the verifier
+	Type() string
```

Comment:
Same as above. Is type an enum?

Reply:
We have multiple verifiers and external plugins so this might not be an enum value

Reply:
Nope, type is a string which represents a specific implementation of verifier. And ratify users could also develop their own verifier plugin with customized type.